tests/conftest.py

```diff
@@ -51,6 +56,21 @@ def all_packages():
     return all_packages
 
 
+@pytest.fixture
+def package_with_deployed_instance_of_contract(valid_package, w3):
+    with open("ethpm/assets/v2-packages/safe-math-lib/SafeMathLib.sol") as file:
+        contract_source_code = file.read()
+    compiled_sol = compile_source(contract_source_code)
+    contract_interface = compiled_sol['<stdin>:SafeMathLib']
+    SafeMathLib = w3.eth.contract(abi=contract_interface['abi'], bytecode=contract_interface['bin'])
+    tx_hash = SafeMathLib.constructor().transact()
+    tx_receipt = w3.eth.getTransactionReceipt(tx_hash)
+    address = tx_receipt['contractAddress']
+    print(address)
```

Comment:
@carver I'd expect every time I run this that I'd get a different address for where the contract was deployed. But i'm getting the exact address every time. Am I missing something? Or is that the expected behavior?

Reply:
> every time I run this

Do you mean the same address across test runs? Or the same address in different tests during the same `pytest` run?

Assuming the latter, `w3` (and more importantly, its eth-tester) is reinitialized every time. Since the contract address is based on sending address's nonce, and the nonce is being reset, you will get the same contract address every time.

Reply:
i'm getting the same address across different `pytest` runs . . .

Reply:
@njgheorghita that is expected.  You can check and verify that the `sender/from` and `nonce` for the transaction are the same.  You should only expect different results if you are using a randomized `from` account.  contract address generation logic can be found here: https://github.com/ethereum/py-evm/blob/master/evm/utils/address.py#L12

Reply:
@pipermerriam ahh great, good to know thanks. any harm in simply choosing a random account from the 10 automatically generated accounts? so i can skip seeding it with ether for deployment